api/src/main/java/org/openmrs/api/db/hibernate/AuditableInterceptor.java

```diff
@@ -117,8 +118,8 @@ private boolean setCreatorAndDateCreatedIfNull(Object entity, Object[] currentSt
 	private boolean changeProperties(Object[] currentState, String[] propertyNames, boolean objectWasChanged,
 	        Map<String, Object> propertyValues, Boolean setNullOnly) {
 		
-		for (String property : propertyValues.keySet()) {
-			if (changePropertyValue(currentState, propertyNames, property, propertyValues.get(property), setNullOnly)) {
+		for (Map.Entry<String, Object> e : propertyValues.entrySet()) {
```

Comment:
I don't think this change is of any value since you still need to use the key after, I think what you need to clean up is cases where they key set is retrieved and the keys are never used except for getting their respective values.